projects/04/mult/Mult.asm

```diff
@@ -9,22 +9,22 @@
 // This program only needs to handle arguments that satisfy
 // R0 >= 0, R1 >= 0, and R0*R1 < 32768.
 
-@R2 // set index 2 to 0
+@R2 // set @R2 to 0
 M=0
 
-(LOOP)
-@R0
+@R0 // set @R0 to be counter
 D=M
 
-@END
-D;JEQ // if index 0 = 0 then end loop
+(LOOP)
+D;JEQ // if counter is 0 then end loop
 
 @R1
 D=M
 @R2
 M=D+M // R2 = R1 + R2
 @R0
 M=M-1
+D=M // make D point to counter

```

Comment:
Gotcha! I changed it to `MD=M-1`

Reply:
Quick explanation to why this is possible: There are 3-bits in the C-instruction dedicated to the destination. Because of that, we can select up to 3 (e.g AMD) destinations to store the computed value.